modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ImmutableProcessorChainExecutor.java

```diff
@@ -113,6 +121,9 @@ private void doProcess(CoreEvent event, Consumer<Result> onSuccess, BiConsumer<T
     }
 
     public void execute() {
+      final Function<Context, Context> innerChainCtxMapping =
+          (Function<Context, Context>) ((InternalEvent) event).getInternalParameter(INNER_CHAIN_CTX_MAPPING);
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this unnecessary cast to "Function". ([squid:S1905](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS1905))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7661&issues=AWjOP1yAotIXnmsUjewX&open=AWjOP1yAotIXnmsUjewX)

[ISSUE_KEY]: AWjOP1yAotIXnmsUjewX